tracer/src/Datadog.Trace.SourceGenerators/Configuration/JsonReader.cs

```diff
@@ -0,0 +1,198 @@
+  // <copyright file="JsonReader.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Datadog.Trace.SourceGenerators.Configuration;
+
+/// <summary>
+/// Json reader as we can't really use vendored libraries here, for framework compatibility reasons
```

Comment:
Is this used an run time? Or only at build time to generate code?

If it's at run time, why can't se use the vendored `Newtonsoft.Json`? Or, if build time, we can't use `System.Text.Json`?